internal/secure/client.go

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright 2019 VMWare Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ *******************************************************************************/
+package secure
+
+import (
+	"flag"
+	"github.com/BurntSushi/toml"
+
+	"github.com/edgexfoundry/docker-edgex-mongo/internal/pkg"
+)
+
+type secureConfigurationStruct struct {
+	pkg.BaseConfigurationStruct
+	pkg.SecretStoreInfo
+}
+
+type secureClient struct {
+	Configuration *secureConfigurationStruct
+}
+
+func NewSecureClient() (*secureClient, error) {
+	config, err := LoadConfig("res/security/configuration.toml")
```

Comment:
I wouldn't put the responsibility of loading the configuration into the DbInitializer concrete types. The logic is duplicated in each type and is a co-mingling of concerns. Separating these will allow you to remove the `GetBaseConfig() pkg.BaseConfigurationStruct` function from `DBInitializer`. In main.go you can then write something like
```
cfg, err := internal.pkg.LoadConfig() //Loads the configuration from file
if err != nil {
    return err
}
dbInitClient, err := internal.NewDBInitClient(cfg)
dbInitClient.Populate()
```
Recommend `"res/security/configuration.toml"` be defined in a constant

Reply:
I have moved the abstraction to the moment when we load the configuration and the credentials. (no interface anymore) Once we have the configuration successfully loaded with the credentials inside (taken from the config file or from Vault) , then just populate the databases and their collections without carrying if the app is working in security_secret_store = on /off.